### PR TITLE
Longest common subsequence in go

### PR DESCRIPTION
**Pull request by BeAgarwal**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes, I have read the guidelines.

### Description

The solution to longest common subsequence added.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2437

### CODE REVIEW

**Comment by BeAgarwal** [comment]
I don't know why the deep source is having the error. Maybe because of several main functions in the directory. Please check @HarshCasper

File: Go/dp/longest_common_subsequence.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        	}  
+    	}  
+          
+    return L[m][n];  
+}  
+  
+
+func main() {
+	str1 := "NEOALGO"
+    	str2 := "ALGORITHM"
```
Comment (HarshCasper) [inline Go/dp/longest_common_subsequence.go]: Take dynamic input

File: Go/dp/longest_common_subsequence.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		fmt.Println("Length of LCS is: ", lcs(str2, str1, n, m))
+	}
+}
+
+/*
+TIME COMPLEXITY: O(M*N)
+SPACE COMPLEXITY: O(M*N)
+OUTPUT
+Length of LCS is:  4
+*/
```
Comment (HarshCasper) [inline Go/dp/longest_common_subsequence.go]: Add a new line at the EOF

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Open up the DeepSource Dashboard and understand the reason why the CI is failing

**Comment by BeAgarwal** [comment]
I have gone through DeepSource dashboard. It is showing other declarations of the main function. This is because we have main in every file of the same directory. Now, it is okay to merge. @HarshCasper

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**